tests/robustness/validate/validate_test.go

```diff
@@ -54,250 +55,1364 @@ func TestDataReports(t *testing.T) {
 
 func TestValidateWatch(t *testing.T) {
 	tcs := []struct {
-		name    string
-		reports []report.ClientReport
+		name         string
+		config       Config
+		reports      []report.ClientReport
+		eventHistory []model.PersistedEvent
+		expectError  string
 	}{
 		{
-			name: "earlier event after bookmark in separate request",
+			name: "Ordered, Unique - ordered unique events in one response - pass",
+			reports: []report.ClientReport{
+				{
+					Watch: []model.WatchOperation{
+						{
+							Responses: []model.WatchResponse{
+								{
+									Events: []model.WatchEvent{
+										putWatchEvent("a", "1", 2, true),
+										putWatchEvent("b", "2", 3, true),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Ordered, Unique - unique ordered events in separate response - pass",
+			reports: []report.ClientReport{
+				{
+					Watch: []model.WatchOperation{
+						{
+							Responses: []model.WatchResponse{
+								{
+									Events: []model.WatchEvent{
+										putWatchEvent("a", "1", 2, true),
+									},
+								},
+								{
+									Events: []model.WatchEvent{
+										putWatchEvent("b", "2", 3, true),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Ordered - unordered events in one response - fail",
+			reports: []report.ClientReport{
+				{
+					Watch: []model.WatchOperation{
+						{
+							Responses: []model.WatchResponse{
+								{
+									Events: []model.WatchEvent{
+										putWatchEvent("b", "2", 3, true),
+										putWatchEvent("a", "1", 2, true),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectError: "broke Ordered - events are ordered by revision; an event will never appear on a watch if it precedes an event in time that has already been posted",
```

Comment:
I think we should create new error types in `validate.go` like
```go
var brokenOrderErr = errors.New("...")
```
and use those here. It might get a little cumbersome to maintain the same text in more than one place.

Reply:
Done